### ISSUE TITLE
Data table paging is no longer working

### ISSUE THREAD

**Issue by underdogwonder** [google/google-visualization-issues#2028]
I've used google datatables on almost everything on a website. Now, my paging on a data table isn't working. Nothing happens whenever I click on the 'Next' button. Not sure how to fix it. Site: ggcircuit.com

**Comment by dlaliberte** [comment]
You can fix your immediate problem temporarily by changing your
google.load() calls to load "1.0" instead of "1".  But, this workaround
will only good until this coming Monday when we push the candidate release
for v42 to become the stable release.

I can't tell yet what is interfering with the page handling, but there have
been some changes in paging options that may be relevant.

Digging into your code a bit, it looks like you have several non-standard
things going on that could be causing problems.  I'm surprised it works at
all, actually.  I see your page related options looking like this:

```
{
                  page: ['enable'],
                  pageSize: [15],
                  pagingSymbols: {
                      prev: ['< PREV'],
                      next: ['NEXT >']
                    },
```

   ...
  }

Primarily, you have arrays where they should just be values.    Try
changing that first.  The pagingSymbols option is not documented, though it
should continue to work, if you specify the values correctly.

If all else fails, you can switch to loading the frozen version 41 instead.

On Tue, Aug 18, 2015 at 4:24 PM, underdogwonder notifications@github.com
wrote:

> I've used google datatables on almost everything on a website. Now, my
> paging on a data table isn't working. Nothing happens whenever I click on
> the 'Next' button. Not sure how to fix it. Site: ggcircuit.com
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/google/google-visualization-issues/issues/2028.

## 

Daniel LaLiberte https://plus.google.com/100631381223468223275?prsrc=2  -
978-394-1058
dlaliberte@Google.com dlaliberte@google.com   5CC, Cambridge MA
daniel.laliberte@GMail.com daniel.laliberte@gmail.com 9 Juniper Ridge
Road, Acton MA

**Comment by ordinaryman** [comment]
Why is there such a change behavior for the version number, contradicting what is provided in the docs?

> The 'visualization' string refers to the google.visualization library, which defines the core utility classes and functions. The '1' specifies which version of Google Charts to load:
> 
> 1 or 1.0 is always the current stable version.
> 1.1 is always the newest release.*

From : https://developers.google.com/chart/interactive/docs/basic_load_libs

![image](https://cloud.githubusercontent.com/assets/6123035/9351151/a6057a24-4674-11e5-92f9-60437f809ccb.png)

I have been using this for a lot of projects as per the document and will have to update from "1" to "1.0" to enable the navigation again!

**Comment by dlaliberte** [comment]
The "current stable version" can and does change over time.  It is the version we consider stable based on testing. 

This time around, for this pre-production testing period, before the actual production release of v42, we did change what loading 1 means, since now 1 is not the same as 1.0 but is the same as 1.1.  But note that if we didn't do this new pre-production testing, you probably wouldn't have found out about problems until we had pushed 1.1 to be the new 1.0.

Changing 1 to 1.0 is only a temporary workaround since it will only avoid the inevitable until this coming Monday when 1.1 becomes the new 1.0.

If you want to avoid being affected by any further changes, you should shift to the frozen version.

**Comment by ordinaryman** [comment]
I understand that using "latest" version will involve having to make changes at some point; what I did not understand was why "1" and "1.0" were not treated the same, as one might expect when reading the documentation. 

> This time around, for this pre-production testing period, before the actual production release of v42, we did change what loading 1 means, since now 1 is not the same as 1.0 but is the same as 1.1.

Now as per your reply, I see that this change was intentional.

> But note that if we didn't do this new pre-production testing, you probably wouldn't have found out about problems until we had pushed 1.1 to be the new 1.0.

Not sure if this would be of much help. What if the users had used "1.0" from the start, since they were both meant to mean the same? They would be in for a surprise when you roll out this change. Only users who had used "1", get to update the version to "1.0" temporarily.

Will use the frozen version as per your recommendation. Thanks.

**Comment by dlaliberte** [comment]
Indeed, it is true that those who already load '1.0' would only see the changes once we go to production. But our experience is that too few people are testing 1.1 ever, so _we_ don't find out about many problems until we go to production, which is way too late.  Rolling back a production release is just as time consuming as pushing out the production release and causes more chaos in the process.

Hopefully, the frozen versions will reduce the need for rollbacks and allow us to proceed with developing the next version.

**Comment by ordinaryman** [comment]
By the way, I narrowed the issue to be due to "pagingButtonsConfiguration":"auto" configuration. This also seems to be the reason for another issue filed recently.
https://github.com/google/google-visualization-issues/issues/2035

I believe this configuration earlier supported "none" / "auto". If you can handle the change related to this, to just ignore this configuration and possibly display a warning below the table/chart, asking the user to communicate to the developer that a configuration option has changed, then you could get more users using and migrating to the latest versions. Thanks.

**Comment by underdogwonder** [comment]
Thank you guys so much for your quick turnaround and insight. Much
appreciated!!!
